openwisp_users/tests/test_admin.py

```diff
@@ -201,6 +201,105 @@ def test_operator_changelist_superuser_column_hidden(self):
+            'is_admin': False,
+            'user': operator}
+        self._create_org_user(**options1)
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_user_change',
+                                           args=[operator.pk]))
+        self.assertContains(response, 'selected>operator-org1</option>')
+        self.assertContains(response, 'selected>operator-org2</option>')
+
+    def test_operator_can_see_organization_add_user(self):
+        org1 = self._create_org(name='operator-org1')
+        org2 = self._create_org(name='operator-org2')
+        operator = self._create_operator()
+        org_permissions = Permission.objects.filter(codename__endswith='organization_user')
+        operator.user_permissions.add(*org_permissions)
+        options1 = {
+            'organization': org1,
+            'is_admin': True,
+            'user': operator}
+        options2 = {
+            'organization': org2,
+            'is_admin': False,
+            'user': operator}
+        self._create_org_user(**options1)
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_user_add'))
+        self.assertContains(response, 'operator-org1</option>')
+        self.assertNotContains(response, 'operator-org2</option>')
+
+    def test_operator_change_organization(self):
+        org1 = self._create_org(name='test-org1')
+        org2 = self._create_org(name='test-org2')
+        default_org = Organization.objects.get(name='default')
+        operator = self._create_operator()
+        org_permissions = Permission.objects.filter(codename__endswith='change_organization')
+        operator.user_permissions.add(*org_permissions)
+        options1 = {
+            'organization': org1,
+            'is_admin': True,
+            'user': operator
+        }
+        options2 = {
+            'organization': org2,
+            'is_admin': False,
+            'user': operator
+        }
+        self._create_org_user(**options1)
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[org1.pk]))
+        self.assertContains(response,
+                            '<input type="text" name="name" value="{0}"'.format(org1.name))
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[default_org.pk]))
+        self.assertContains(response,
+                            '<input type="text" name="name" value="{0}"'.format(default_org.name))
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[org2.pk]))
+        if response.status_code == 200:
```

Comment:
I don't understand, why there's an if here? Shouldn't a test expect only one outcome?

Reply:
Yes that's true but on django>=2.1, 
```response = self.client.get(reverse('admin:openwisp_users_organization_change',
                                           args=[org2.pk]))```
returns a 200 status code whereas in django<2.1, it returns  a status code of 403 which caused travis to fail if i just used the expected `self.assertContains`  test. Thus, i added the if-else to take care of travis while making sure the required test is performed during each case.